File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i=0,j=0,n=s.size(),maxi=INT_MIN;
        if(n==0){
            return 0;
        }
        set<char> newt;
        while(j<n){
            if(newt.find(s[j])==newt.end()){
                newt.insert(s[j]);
                maxi= max(maxi, j-i+1);
                j++;
            }
            else{
                newt.erase(s[i]);
                i++;
            }
        }
        return maxi;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i,j=0,0
        n= len(s)
        res= []
        maxi= 0
        while i<=j and j<n:
            if s[j] not in res:
                res.append(s[j])
                maxi= max(maxi, len(res))
                j+=1
            else:
                res.remove(s[i])
                i+=1
        return maxi
